ui/raidboss/data/06-ew/raid/p2n.ts

```diff
@@ -0,0 +1,206 @@
+          data.bodyActor = undefined;
+          return;
+        }
+        data.bodyActor = hippos[0];
+      },
+      alertText: (data, matches, _output) => {
+        if (!data.bodyActor) {
+          console.error('SpokenCataract: No boss actor found. Did the promise fail?');
+          return;
+        }
+        const heading = data.bodyActor?.Heading;
+        let bodyHeading = null;
+
+        if (heading < -3.0 || heading > 3.0)
+          bodyHeading = 'north';
+        if (heading < -1.56 && heading > -1.58)
+          bodyHeading = 'west';
+        if (heading > -0.1 && heading < 0.1)
+          bodyHeading = 'south';
+        if (heading > 1.56 && heading < 1.58)
+          bodyHeading = 'east';
+
+        if (matches.id === '67F8') {
+          if (bodyHeading === 'north')
+            return 'body north, head south';
+          if (bodyHeading === 'west')
+            return 'body west, head east';
+          if (bodyHeading === 'south')
+            return 'body south, head north';
+          if (bodyHeading === 'east')
+            return 'body east, head west';
+        }
+        if (matches.id === '67F7') {
+          if (bodyHeading === 'north')
+            return 'body north, head east';
+          if (bodyHeading === 'west')
+            return 'body west, head north';
+          if (bodyHeading === 'south')
+            return 'body south, head west';
+          if (bodyHeading === 'east')
+            return 'body east, head south';
+        }
+        if (matches.id === '67F9') {
+          if (bodyHeading === 'north')
+            return 'body north, head west';
+          if (bodyHeading === 'west')
+            return 'body west, head south';
+          if (bodyHeading === 'south')
+            return 'body south, head east';
+          if (bodyHeading === 'east')
+            return 'body east, head north';
+        }
+      },
+      outputStrings: {
+        text: {
+          en: 'Tank Laser on YOU',
+          de: 'Tank Laser auf DIR',
+          fr: 'Tank laser sur VOUS',
+          ja: '自分にタンクレーザー',
+          cn: '坦克射线点名',
+          ko: '탱 레이저 대상자',
+        },
+      },
+    },
+    {
+      id: 'P2N Sewage Deluge',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '67F6', source: 'Hippokampos', capture: false }),
+      alertText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'Aoe get on grid',
+        },
+      },
+    },
+    {
+      // Spread aoe marker on some players, not all
+      id: 'P2N Tainted Flood',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '6809', source: 'Hippokampos', capture: true }),
+      condition: (data, matches) => matches.target === data.me,
+      response: Responses.spread(),
+    },
+    {
+      // Drops aoe zones beneath you -> run to dodge (on everyone)
+      // Not working currently properly -> On you not working.
```

Comment:
I'm not completely sure I understand what this means. In my logs, it looks like Hippokampus casts Sewage Eruption on itself, and then casts it at unknown entities on the floor under players, not *at* players. Do you have a way to determine which players it is targeting even so? (I ask because that would be useful elsewhere.)

Reply:
This has also been fixed in a later version

Reply:
Fixed in the latest version!